lib/src/components/NxFilterInput/__tests__/NxFilterInput.test.tsx

```diff
@@ -5,160 +5,108 @@
-import { mount, shallow } from 'enzyme';
-import { faFilter, faSearch } from '@fortawesome/free-solid-svg-icons';
+import { waitFor } from '@testing-library/react';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+import userEvent from '@testing-library/user-event';
 
 import NxFilterInput, { Props } from '../NxFilterInput';
-import { PrivateNxTextInput } from '../../NxTextInput/NxTextInput';
-import NxFontAwesomeIcon from '../../NxFontAwesomeIcon/NxFontAwesomeIcon';
-import NxButton from '../../NxButton/NxButton';
-import Close from '../../../icons/Close';
 
 describe('NxFilterInput', function() {
-  const minimalProps = { value: '' },
-      shallowComponent = enzymeUtils.getShallowComponent<Props>(NxFilterInput, minimalProps),
-      mountedComponent = enzymeUtils.getMountedComponent<Props>(NxFilterInput, minimalProps);
-
-  it('renders a PrivateNxTextInput with the nx-filter-input class', function() {
-    const component = shallowComponent();
-    expect(component).toHaveClassName('nx-filter-input');
-    expect(component).toMatchSelector(PrivateNxTextInput);
+  const minimalProps: Props = { value: '' },
+      quickRender = rtlRender(NxFilterInput, minimalProps),
+      renderEl = rtlRenderElement(NxFilterInput, minimalProps);
+
+  it('renders an Input', function() {
+    expect(quickRender().getByRole('textbox').tagName).toBe('INPUT');
   });
 
-  it('adds custom classnames as specified', function() {
-    const component = shallowComponent({ className: 'foo' });
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-filter-input');
+  it('adds additional specified classNames to the top-level element', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
+
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('passes misc props to the PrivateNxTextInput', function() {
-    const onChange = jest.fn(),
-        onKeyPress = jest.fn(),
-        component = shallowComponent({
-          value: 'foo',
-          onChange,
-          onKeyPress,
-          id: 'bar'
-        });
-
-    expect(component).toHaveProp('value', 'foo');
-    expect(component).toHaveProp('onChange', onChange);
-    expect(component).toHaveProp('onKeyPress', onKeyPress);
-    expect(component).toHaveProp('id', 'bar');
+  it('passes additional attrs to the Input', function() {
+    const input = quickRender({ id: 'foo', lang: 'en-US' }).getByRole('textbox');
+
+    expect(input).toHaveAttribute('id', 'foo');
+    expect(input).toHaveAttribute('lang', 'en-US');
   });
 
-  it('does not pass validatable, validationErrors, or type props to the PrivateNxTextInput', function() {
-    const component = shallowComponent({
+  it('does not pass validatable, validationErrors, or type props to the Input', function() {
```

Comment:
Testing that `type` is ignored in this way is valid, but for the others they wouldn't have become attributes on the `<input>`  anyway, so I think a different approach is needed.  Maybe check that `aria-invalid` doesn't get set to true, that `aria-errormessage` doesn't get set, and that the `validationErrors` text doesn't show up anywhere in the rendered DOM.

Reply:
Updated here 67f47c9c38f4b270233459595b587866d52e5173. Thanks.
For the `aria-errormessage`, it is set with a unique ID if not disabled in `NxTextInput`. 
So there is `aria-errormessage` when valid, seems like it is correct as long as the element referenced by the `aria-message` is hidden according to the WAI-ARIA doc https://www.w3.org/TR/wai-aria-1.2/#aria-errormessage 

> When an object is in a valid state, it has either "aria-invalid" set to false or it does not have the "aria-invalid" attribute. Authors MAY use aria-errormessage on an object that is currently valid, but only if the element referenced by aria-errormessage is "hidden", because the message it contains is not pertinent.

Reply:
> For the aria-errormessage, it is set with a unique ID if not disabled in NxTextInput.
So there is aria-errormessage when valid, seems like it is correct as long as the element referenced by the aria-message is hidden according to the WAI-ARIA doc https://www.w3.org/TR/wai-aria-1.2/#aria-errormessage

OK, looking closer what happens is that the `aria-errormessage` attribute is set but the element that it refers to is not present in the DOM.  I'm not totally sure that's valid, but our a11y tests seem to be happy with it so I guess it is.  At any rate, the jest RTL matchers include a [`toHaveErrorMessage`](https://github.com/testing-library/jest-dom#tohaveerrormessage) matcher which I think would be good to use here `(with a `not`)

Reply:
> OK, looking closer what happens is that the aria-errormessage attribute is set but the element that it refers to is not present in the DOM. I'm not totally sure that's valid, but our a11y tests seem to be happy with it so I guess it is.

This is out of scope of this ticket, but do you think it might be much more clear if we apply the `aria-errormessage` attr to the input only when it's invalid?

> At any rate, the jest RTL matchers include a [toHaveErrorMessage](https://github.com/testing-library/jest-dom#tohaveerrormessage) matcher which I think would be good to use here (with a not`)

Thanks for the suggestion, with this matcher I think I don't need the query of alert and error text, updated here 914b2f7062be0607d25692deaa454124621921a1.

Reply:
> but do you think it might be much more clear if we apply the aria-errormessage attr to the input only when it's invalid?

A little bit maybe, but what we have currently isn't really wrong according to the a11y tests.